java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java

```diff
@@ -63,6 +63,93 @@
  */
 public class JdbcToArrow {
 
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |          Partial Convert API                        |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+  /**
+   * For the given JDBC {@link ResultSet}, fetch the data from Relational DB and convert it to Arrow objects.
+   * Note here uses the default targetBatchSize = 1024.
+   *
+   * @param resultSet ResultSet to use to fetch the data from underlying database
+   * @param allocator Memory allocator
+   * @return Arrow Data Objects {@link ArrowVectorIterator}
+   * @throws SQLException on error
+   */
+  public static ArrowVectorIterator sqlToArrowVectorIterator(
+      ResultSet resultSet,
+      BaseAllocator allocator)
+      throws SQLException, IOException {
+    Preconditions.checkNotNull(allocator, "Memory Allocator object can not be null");
+
+    JdbcToArrowConfig config =
+        new JdbcToArrowConfig(allocator, JdbcToArrowUtils.getUtcCalendar());
+    return sqlToArrowVectorIterator(resultSet, config);
+  }
+
+  /**
+   * For the given JDBC {@link ResultSet}, fetch the data from Relational DB and convert it to Arrow objects.
+   * Note if not specify {@link JdbcToArrowConfig#targetBatchSize}, will use default value 1024.
+   * @param resultSet ResultSet to use to fetch the data from underlying database
+   * @param config    Configuration of the conversion from JDBC to Arrow.
+   * @return Arrow Data Objects {@link ArrowVectorIterator}
+   * @throws SQLException on error
+   */
+  public static ArrowVectorIterator sqlToArrowVectorIterator(
+      ResultSet resultSet,
+      JdbcToArrowConfig config)
+      throws SQLException, IOException {
+    Preconditions.checkNotNull(resultSet, "JDBC ResultSet object can not be null");
+    Preconditions.checkNotNull(config, "The configuration cannot be null");
+    return ArrowVectorIterator.create(resultSet, config);
+  }
+
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |           Deprecated API                        |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+  /**
```

Comment:
this makes the diff harder to follow, can the be put back in there original spots?

Reply:
it  looks like these might still be out of place

Reply:
Sorry, fixed now.